CONTRIBUTING.md

```diff
@@ -0,0 +1,91 @@
+# Contributing
+
+👍🎉 First off, thank you for taking the time to contribute! 🎉👍
+
+The following is a set of guidelines for contributing. These are just guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+## What Should I Know Before I Get Started?
+
+### Code of Conduct
+
+This project adheres to the [Contributor Covenant](./code-of-conduct.md). By participating, you are expected to uphold this code.
+
+Please report unacceptable behavior to one of the [Code Owners](./CODEOWNERS).
+
+### How Do I Start Contributing?
+
+The below workflow is designed to help you begin your first contribution journey. It will guide you through creating and picking up issues, working through them, having your work reviewed, and then merging.
+
+Help on open source projects is always welcome and there is always something that can be improved. For example, documentation (like the text you are reading now) can always use improvement, code can always be clarified, variables or functions can always be renamed or commented on, and there is always a need for more test coverage. If you see something that you think should be fixed, take ownership! Here is how you get started:
+
+## How Can I Contribute?
+For any contributions that need design changes/API changes, first contribute an ADR. Reason for ADR: teams agree on the design, to avoid back and forth after writing code. An ADR gives context on the code being written.
```

Comment:
```suggestion
## TODO: Add link to ADR  and add template on this roadmap

For any contributions that need design changes/API changes, first contribute an ADR. Reason for ADR: teams agree on the design, to avoid back and forth after writing code. An ADR gives context on the code being written.
```